### ISSUE TITLE
Text ending in a hash-character incorrectly detected as closed heading

### ISSUE THREAD

**Issue by mtausig** [DavidAnson/markdownlint#404]
I have a document with some code snippets, one of them being `C#`:

```
# Main

## Samples

### Java

foo

### C#

bar
```

The problem is, that this triggers rules MD003 and MD020 because the validator believes that I am trying to use a closed heading style in the last heading.

Demo sample: https://dlaa.me/markdownlint/#%25m%23%20Main%0A%0A%23%23%20Samples%0A%0A%23%23%23%20Java%0A%0Afoo%0A%0A%23%23%23%20C%23%0A%0Abar%0A

**Comment by nschonni** [comment]
Yes, you need to escape the trailing `#` as `### C\#`

**Comment by DavidAnson** [comment]
Agreed. Here's your sample with that change: https://dlaa.me/markdownlint/#%25m%23%20Main%0A%0A%23%23%20Samples%0A%0A%23%23%23%20Java%0A%0Afoo%0A%0A%23%23%23%20C%5C%23%0A%0Abar%0A

**Comment by mtausig** [comment]
Thank you. Didn't know that.
